src/plugins/ui-plugin.ts

```diff
@@ -18,10 +19,7 @@ export default class UIPlugin extends EventEmitter {
 
     render() {}
```

Comment:
By the way, it is desirable to make all methods abstract like setSettings.)

Reply:
Hm, typescript complains that if method is marked as abstract, then it must be always implemented by child, so it does not care if I mark it as optional

Reply:
Have you tried that code snippet I mentioned earlier?) https://github.com/pyatyispyatil/flame-chart-js/pull/25#discussion_r841827012

Method should be described like that `optional?(): string;` without `abstract`

Reply:
Yes, setSettings is marked as optional
but you said `it is desirable to make all methods abstract like setSettings` and do you mean to add `abstract` keyword in front of the methods or to just mark them as optional?
I think we are maybe not talking about the same thing :D

Reply:
ok It was a misunderstanding then :D
will do the changes

Reply:
Oh yes, sorry.  I meant only `optional?(): string` and it is abstract method, because if it is without a body, then the method is abstract.)